perun/utils/view_helpers.py

```diff
@@ -0,0 +1,206 @@
+            },
+        }
+    )
+
+
+def add_y_units(profile_header: MutableMapping[str, Any], of_key: str, y_axis_label: str) -> str:
+    """Add units to Y axis label if the Y dimension has one.
+
+    :param profile_header: the header part of a profile.
+    :param of_key: the Y dimension name.
+    :param y_axis_label: the current Y dimension label.
+
+    :returns: the Y dimension label with a unit, if eligible.
+    """
+    if of_key in ("count", "nunique") or y_axis_label.endswith("]"):
+        # Skip if the label already has a unit or the of_key doesn't have a unit
+        return y_axis_label
+    # The of_key should have a unit
+    profile_type = profile_header["type"]
+    type_unit = None
+    try:
+        # Check if we have a 'unit' corresponding to the profile type
+        type_unit = profile_header["units"][profile_type]
+    except KeyError:
+        # The profile type might be called a bit differently in the 'units', e.g.,
+        # 'mixed' -> 'mixed(time delta)'
+        for unit_name, unit in profile_header["units"].items():
+            if profile_type in unit_name:
+                type_unit = unit
+                break
+    return y_axis_label if type_unit is None else f"{y_axis_label} [{type_unit}]"
+
+
+def save_view_graph(
+    graph: Union[hv.Chart, hv.Layout, hv.Overlay],
+    filename: str,
+    view_in_browser: bool = False
+) -> None:
+    """Save or view the provided holoviews graph.
+
+    :param graph: a holoviews figure (or their composition) that will be rendered.
+    :param filename: name of the output file, where the graph will be saved.
+    :param view_in_browser: ``True`` if the graph should be displayed straight in the browser.
+    """
+    if view_in_browser:
+        output_file(filename)
+        show(hv.render(graph))
+    else:
+        hv.save(graph, filename)
+
+
+def process_profile_to_graphs(
+    factory_module: ModuleType,
+    profile: Profile,
+    filename: str,
+    view_in_browser: bool,
+    func: str,
+    of_key: str,
+    **kwargs: Any,
+) -> None:
+    """Wrapper function for constructing the graphs from profile, saving it and viewing.
+
+    Wrapper function that takes the factory module, constructs the graph for the given profile,
+    and then saves it in filename and optionally view in the registered browser.
+
+    :param factory_module: module which will create the graph.
+    :param profile: a Perun profile that will be processed.
+    :param filename: output filename for the graph.
+    :param view_in_browser: ``True`` if the created graph should be displayed in registered browser
+           after it is constructed.
+    :param func: function used for aggregation of the data.
+    :param of_key: key that will be aggregated in the graph.
+    :param kwargs: rest of the keyword arguments.
+    :raises AttributeError: when the factory_module is missing some of the functions.
+    """
+    profiles.is_key_aggregatable_by(profile, func, of_key, "of_key")
```

Comment:
Should this be in some `if` or something? Or does it fire exception? Then it should be caught maybe with some error?

Reply:
The function raises an InvalidParameterException that has a suitable user-friendly string representation and has been used to report click parameter errors to the users so far. Catching the error would not make much sense as it would be re-raised anyway. Note that this was taken as-is from the original boken_helpers.py.

Reply:
Yeah, I get it some parts are taken from previous code, and might be caused by me :D. I wonder if it is worth either (1) adding comment `# this raises exception` or (2) renaming the function? The latter would require some more digging though how it is used (probably rather in `if`s). What do you think?

Reply:
In that case, I'd say let's document it in the docstring (added bonus is that Sphinx supports it and it will appear in the docs). Although the `is_key_aggregatable_by` method has it documented, this method does not mention it.

Reply:
Ook, I consider everything fully discussed. Feel free to merge it yourself, once you fix what we have aggreed upon.

Reply:
I updated the method docstring to mention the exception and its cause. Would that work for you?

Reply:
Yes